### ISSUE TITLE
Pipe(?) operator explaination

### ISSUE THREAD

**Issue by petemcfarlane** [mint-lang/mint#228]
I can't find any information in the docs nor GitHub about the `|>` operator (sometimes called "pipe" in other languages). 
It's not clear to me if this is functional composition or just syntactic sugar for chaining other functions.

Why do I have to write the parenthesis in `"Argument" |> Debug.log()` why can't I write `"Argument" |> Debug.log`?

Why can't I use anonymous functions (unless I'm being a tool)
```
"https://swapi.co/api/planets"
  |> Http.get()
  |> Http.send()
  |> (response: Http.Response) { response.body }
  |> Debug.log()
```

Can I use composition to do something like this (completely useless but basic example)
`String.toLowerCase() |> String.toUpperCase()`?

Maybe we could put something in the Function section?

**Comment by gdotdesign** [comment]
I've transferred this issue to the main repository.

> Why do I have to write the parenthesis in `"Argument" |> Debug.log()` why can't I write `"Argument" |> Debug.log?`

It was a syntax choice from my part, this can be improved of course mainly by leaving out the parentheses.

> Why can't I use anonymous functions (unless I'm being a tool)

This is probably something I overlooked, it is certainly possible to implement.

> Can I use composition to do something like this (completely useless but basic example)
String.toLowerCase() |> String.toUpperCase()?

Yes you can do that.

-----

In general this has not been documented, so I'll keep this issue for that and also for other discussion related to the pipe operator.

**Comment by gdotdesign** [comment]
This issue is addressed in #216

**Comment by gdotdesign** [comment]
Added documentation to https://www.mint-lang.com/guide/reference/functions
